### PR TITLE
Dab

### LINKED ISSUE #9909: Look at my DAB
Dabbing Emote.

There should be a "dab" button that allows players to dab in-game, but has a chance to give brain damage every time you do it.

It would take literally ten seconds to code a verb and implement it, and I think it would be hilarious.

### PR DESCRIPTION
**Pull request by AnthonyJNgo**
:cl:
add: You can dab now, with fun effects on success and failure!
/:cl:

You can now dab! Doing it correctly will make you do a sick dab and make you trip out a bit. Doing it incorrectly will cause you to scream out something and hurt your brain.

*dab

closes #9909  (It took a YEAR for this to be done and I ended up doing it? Thanks Carbon!)

There was a small typo in the gif, I corrected it.

KHlJgu95is

### CODE REVIEW

**Comment by ghost** [comment]
Seems to be an issue in the code. The brain damage won't proc all the time the way it's written.

**Comment by Manly-Dwarf** [comment]
No *smug style animation 0/10

**Comment by Manly-Dwarf** [comment]
also the lines it makes you say are criminally unfunny

**Comment by AnthonyJNgo** [comment]
They're braindamaged, it's supposed to be ironically unfunny.

**Comment by Manly-Dwarf** [comment]
criminally unfunny either way

**Comment by DaydreamPuppeh** [comment]
Why is there no actual dab

**Review by ghost — APPROVED** [review]
dab

**Comment by Manly-Dwarf** [comment]
this is trash don't actually merge this until it can live up to the hype of a dab emote

**Comment by AnthonyJNgo** [comment]
@Manly-Dwarf I refuse to live under this dictatorship, I am obviously being singled out for my lack of artistic ability.

File: hippiestation/code/modules/mob/living/carbon/human/emote.dm [hunk 1]
```diff
@@ -286,4 +286,30 @@
...
+	var/dabstrength = rand(1,3)//there should be a pretty high chance of you getting brain damage from dabbing
+	if (dabstrength == 1)
+		user.visible_message("<span class='warning'><b>[R]</b> tries to dab, but they hit themself in the head on accident!</span>", "<span class='warning'>You try dabbing but you fuck it up somehow!</span>")
+		playsound(R, 'sound/effects/snap.ogg', 50, TRUE)
+		R.adjustBrainLoss(20)
+		new armblood(R.loc)
+		return
+	R.visible_message("<span class='warning'><b>[user]</b> does an extremely cool dab! Yeet! </span>", "<span class='warning'>Holy shit, you just did a sick dab!</span>")
+	R.hallucination += 2
+	R.say(pick("CARBON COME BACK PLS!!","WE DON'T EVEN NEED MAINTAINERS!","HIGHPOP BY AUGUST!!"))
```
Comment (actioninja) [inline hippiestation/code/modules/mob/living/carbon/human/emote.dm:314]: based

Comment (ghost) [reply hippiestation/code/modules/mob/living/carbon/human/emote.dm:314]: based

Comment (Dinopeaks) [reply hippiestation/code/modules/mob/living/carbon/human/emote.dm:314]: Based

**Comment by actioninja** [comment]
I made a dab emote a while ago that tyrannical fruit rejected, among other things it ripped off your arms, could cause special brain traumas, and could break your legs.

**Comment by ghost** [comment]
pr it actioninja

**Comment by AnthonyJNgo** [comment]
@actioninja @zamolxius Alright I could flesh it out a bit and add some of those things onto it, if it will appease @manly-dwarf

**Comment by Manly-Dwarf** [comment]
show it off actioninja

**Comment by actioninja** [comment]
I deleted the branch which had some fixes on it but this is roughly it https://github.com/tgstation/tgstation/pull/43682/files
restraint check doesn't work and you could still do it while cuffed or while missing arms to rip your arms off and escape cuffs.

**Comment by ghost** [comment]
seems like a fair tradeoff

**Comment by Manly-Dwarf** [comment]
we need to find a good dab sprite and make it appear like the smug emote

**Comment by karmaisblackandbluepilled** [comment]
-1 because of the braindamage

**Comment by Maroonenahmoron** [comment]
No animation, no satisfaction

File: hippiestation/code/modules/mob/living/carbon/human/emote.dm [hunk 2]
```diff
@@ -286,4 +286,30 @@
...
+	message = "dabs!"
+
+/datum/emote/living/dab/run_emote(mob/living/carbon/user, params)
+	if(!ishuman(user))
+		return ..()
+	var/obj/item/bodypart/r_arm/D = user.get_bodypart(BODY_ZONE_R_ARM)
+	var/mob/living/carbon/human/R = user
+	if(!D)
+		to_chat(user, "<span class='warning'>You can't dab without your right arm!</span>") //I'm right handed, so we're making it rely on the right arm :smug:
+		return FALSE // This will tell you that the emote is unusable as well.
```
Comment (ghost) [inline hippiestation/code/modules/mob/living/carbon/human/emote.dm:303]: the 'Unusable Emote' message is very annoying and should not be there. Fix that and I'll merge.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by yoyobatty** [comment]
Stale.

**Pull request closed without merging**